### PR TITLE
Introduces docz site generation

### PR DESCRIPTION
**Pull request by marcgarreau**
### What does it do?
- Introduces [docz](https://www.docz.site/docs/introduction) static site generation for any shared documentation between EF JS teams.
- One page is included listing the four projects. Uses pure markdown, but the `.mdx` file supports JSX, allowing arbitrary React components to be used.
- To try it out locally: `yarn && yarn docz:dev` will install deps and open a browser tab with the site.
- Quick explanation of some other features I like: https://github.com/ethereum/js-organization/issues/4#issuecomment-562769909
- Merging this would not deploy the site anywhere yet; `yarn docz:build` produces static assets which could then be deployed.
### Relevant Issues?
https://github.com/ethereum/js-organization/issues/4
### Screenshots?
Default theme - light:
Screen Shot 2019-12-06 at 4 21 44 PM
Default theme - dark:
Screen Shot 2019-12-06 at 4 23 40 PM

### CODE REVIEW

**Comment by nivida** [comment]
@marcgarreau Looks great! Is it possible to merge this into the ``gh-pages`` branch and to generate the github page with a github action as soon as the ``gh-pages`` branch got updated?

File: package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "js-organization",
+  "version": "1.0.0",
```
Comment (holgerd77) [inline package.json]: Can we start with a lower version number here, so `0.1.0` or even `0.0.1` and keep the `1.0.0` for a first version where structured has been relatively settled and we have got something which we would publish officially?

**Comment by holgerd77** [comment]
@marcgarreau @nivida `gh-pages` branch is probably a good idea and to build automatically.

**Name**: The change of the team name is a pretty big thing. We can discuss this if there is demand, but let's not mix this in here since it is unrelated to the documentation setup.

**Domain**: I would also keep this for later until we decide to go more public with the documentation. We might also be able to use a subdomain of the new EF website - already made some first enquiry here - not completely clear yet though.

File: package.json (hunk 1 above)
Comment (marcgarreau) [reply package.json]: 👍 dropped to 0.0.1 as we have nothing published yet

**Comment by marcgarreau** [comment]
@holgerd77 I don't have permissions to do it, but if you update the source of GitHub Pages to `gh-pages`, it should result in a deploy to check out on https://ethereum.github.io/js-organization ...hopefully.

Screen Shot 2019-12-09 at 3 02 55 PM

A new script, `yarn deploy`, will build the files and deploy the output to the `gh-pages` branch, then GitHub will do the rest. I didn't go as far as setting up a GitHub Action to automate the deploy on push to master, for example, because I'm unsure of the financial implications.

**Comment by holgerd77** [comment]
@marcgarreau this is actually set to gh-pages, probably another reason? Maybe deploying is disallowed on the organization level? Just a guess though.

**Comment by marcgarreau** [comment]
🤔 odd, I deployed to my fork with what I believe to be the same configuration: https://marcgarreau.github.io/js-organization/. I'll double-check that I didnt leave anything out and ping @jpitts to check on the org-level policy theory.

**Comment by evertonfraga** [comment]
It's only published when admins make the changes ([reference](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#troubleshooting-publishing-problems-with-your-github-pages-site
)).

> _"Note: If your site's source files are located in the default publishing source—master for user and organization sites or gh-pages for project sites—but your site has not published automatically, **make sure someone with admin permissions and a verified email address** has pushed to the default publishing source."_

**Comment by marcgarreau** [comment]
That would explain it - thanks @evertonfraga. Looks like its on you @holgerd77 to pull and run the `yarn deploy` command if you want to test it.

**Comment by marcgarreau** [comment]
The site is already deployed - though I think we've agreed not to make any effort to distribute the link for now. I'm working on more content now, but are you okay with merging this checkpoint @holgerd77?

**Comment by evertonfraga** [comment]
Not sure if this is the right place, but this is a cool gadget to help with publishing the built website in `gh_pages`. 
https://github.com/marketplace/actions/push-git-subdirectory-as-branch

**Comment by marcgarreau** [comment]
That does look handy and more streamlined @evertonfraga. I'm still unclear if its okay to use Actions in this org though.

Side note: would also be nice to set up deploy previews, like we've done with netlify for the devcon site in the past. Its a small pain to test updates now, using my fork.

**Comment by evertonfraga** [comment]
Right, I configured Netlify for this repo. Will make a test PR to test the "deploy preview" feature.
https://ethereum-js-organization.netlify.com

**Comment by evertonfraga** [comment]
Deploy preview works: 
https://5df7e018815390000aedc788--ethereum-js-organization.netlify.com

**Comment by holgerd77** [comment]
Have up-leveled the merge requirements here so that we are in line with other repos and standards and there is now a review needed for each PR (doesn't need to be me of course).

From my side we can merge this here. Didn't completely follow your last discussions and not sure if there is now something you guys want to add so I'll approve and leave merging to you.

**Comment by marcgarreau** [comment]
@evertonfraga Thanks for taking the initiative and setting up previews. Looks like the deploy has a familiar bug though: some assets have a path issue and aren't picked up (e.g. dark mode toggle does nothing, nav links broken). This is addressed with the `PUBLIC_URL` in the deploy script, but we'll have to figure out something more dynamic for previews. This issue is separate from this PR, so will merge and address next.

**Pull request merged**